source/Missions/MissionModule.cs

```diff
@@ -52,12 +58,17 @@ protected override void Load(ContainerBuilder builder)
 
             // Interface classes
             builder.RegisterType<LiteNetP2PClient>().As<INetwork>().AsSelf().InstancePerLifetimeScope();
-            
+
             builder.RegisterType<RandomEquipmentGenerator>().As<IRandomEquipmentGenerator>();
             builder.RegisterType<PacketManager>().As<IPacketManager>().InstancePerLifetimeScope();
             builder.RegisterType<EventPacketHandler>().AsSelf().InstancePerLifetimeScope();
             builder.RegisterType<MovementHandler>().AsSelf().InstancePerLifetimeScope();
-            
+            builder.RegisterType<MissileHandler>().As<IMissileHandler>();
+            builder.RegisterType<WeaponPickupHandler>().As<IWeaponPickupHandler>();
+            builder.RegisterType<WeaponDropHandler>().As<IWeaponDropHandler>();
+            builder.RegisterType<ShieldBreakHandler>().As<IShieldBreakHandler>();
+            builder.RegisterType<AgentDamageHandler>().As<IAgentDamageHandler>();
```

Comment:
All interfaces here need to be added to the `CoopArenaController` constructor as a parameter.

This is needed so that when the `CoopArenaController` is created, these objects are also created by the DI framework.

I.E.
```CS
CoopArenaController(
    INetworkMessageBroker networkMessageBroker,
    INetworkAgentRegistry agentRegistry, 
    IRandomEquipmentGenerator equipmentGenerator,
    IBinaryPackageFactory packageFactory,
    IMissileHandler missileHandler,
    IWeaponDropHandler weaponDropHandler,
    ...)
```